index.js

```diff
@@ -34,6 +33,29 @@ function load() {
   return result;
 }
 
+function extend(a, b) {
+  // iterate over all direct and inherited enumerable properties
+  for (const name in b) {
+    let newValue = b[name];
+
+    // check if the new value is an object and its property exists on the former
+    if (typeof newValue === 'object' && newValue !== null && name in a) {
+      const oldValue = a[name];
+
+      // check if the former value is also an object
+      if (typeof oldValue === 'object' && oldValue !== null) {
+        // update the new value as a new object extended by the former and later
+        newValue = extend(extend({}, oldValue), newValue);
```

Comment:
Are two `extend` calls needed here? Since `extend` modifies `a` in-place, I think this could be just `extend(oldValue, newValue)`, assuming the `a[name] = newValue` is done in an else branch and not unconditionallly.

Reply:
Yes, I agree that this is **peculiar**, which is a reason I think it belongs _in_ the codebase and not abstracted away.

From my PR, with emphasis added:

> ... it extends them in a unique way, where any two nested objects are merged into one **new** object

The (jQuery) extend method mutates a root object, but it does not mutate any of its nested objects.

If this (new) extend method were simplified so that the nested objects were also mutated, then many tests would fail.

From those failures I inferred that this **particular** functionality is by design.

Reply:
I defer to @ddbeck, but I'd say that as long as the resulting object is identical to before, various peculiarities of `extend` that we don't depend on need not be preserved, indeed it's better if they don't work :)

Reply:
> I defer to @ddbeck, but I'd say that as long as the resulting object is identical to before, various peculiarities of `extend` that we don't depend on need not be preserved, indeed it's better if they don't work :)

I agree with this. I'm fine with making this less weird, if the resulting object is the same or it breaks in ways that reveal genuine data problems (as in #9842).

I think `extend` was originally adopted when BCD was quite immature and it was convenient for experimentation, particularly when the schema went through some major changes early on (before my time—@Elchi3 can speak to this history better than I can). I don't think we've ever actually benefited from any of `extend'`s oddities.

So I guess what I'm saying here is: feel free to make this less permissive, but let's make sure to do a full accounting of any differences in the resulting object before we merge.

Reply:
As it is, the resulting objects are identical before and after this change, and also if https://github.com/mdn/browser-compat-data/pull/9842 is merged, my shorter and stricter `extend` produces the exact same result.

This should be confirmed once more before merging, of course.